IaC project structure. Team working using Terraform

Durf1985 opened the pull request:
## Organizing infrastructure code and working on infrastructure in a team

## In the process done:
* Import of existing infrastructure into terraform configuration (firewall rule)
* Splitting the packer template into 2 files to create two VMs (application and database)
* Creating the above instances. 
* Allocation of the remaining infrastructure in the main file.tf by individual files
* Creating `db`, `app`, `vpc` modules
* Parameterization of modules
* Creating Stage and Prod environments
* Import and configure storage-bucket
* Setting up the storage of the status file in the remote backend
* Configuring the connection between Application VM and data base VM

## How to launch a project:
Run next commands first in the `your_repo/terraform` folder, and then in the `your_repo/terraform/stage` and `your_repo/terraform/prod` folders, respectively

```bash
terraform init
terraform plan
terraform apply
```
On the virtual machine containing the database, edit the configuration file `mongod.conf`
and add the internal IP address of your virtual machine with the application to it

## How to check the performance:
In the browser address bar, enter the external Ip address of your Application VM
```text
http://external_ip_address_reddit_app_vm_instance:9292
```
## PR checklist

The label with the homework number, in label description - homework lecture and topic

Durf1985 committed:
DO-57 SSH, Import, External IP
Create ssh firewall rule. Import resources. Create external IP resource

loktionovam commented on terraform/terraform.tfvars.example:
```diff
@@ -1,6 +1,6 @@
-project = "clgcporg2-067"
+project = "clgcporg2-092"
```
here is an example file so you must not change any "example" value to the real existing

loktionovam commented on terraform/terraform.tfvars.example:
```diff
@@ -1,6 +1,6 @@
 private_key_path = "~/.ssh/appuser"
 count_vm = 2
-source_image = "projects/clgcporg2-067/global/images/reddit-full-1665389936"
+source_image = "projects/clgcporg2-092/global/images/reddit-full-1665389936"
```
here is an example file so you must not change any "example" value to the real existing

Durf1985 committed:
DO-58 Split monolith vm onto two: app and db
Created two packer template for mongodb (db.json) and application (app.json)
app.json - without systemd unit, will be added later
Split main.tf for 4 files -> "main.tf" "app.tf" "db.tf" "vpc.tf"
Add into variables.tf new variables

Durf1985 committed:
DO-59 Split config onto modules, parametrize

The modules "app", "db", "vpc" were created, parameterized and loaded
into the main file `your_repo/terraform/main.tf`
Remove app.tf db.tf vpc.tf from terraform folder
Added missing reddit-app launch unit on reddit-app VM instance image

Durf1985 committed:
DO-60 Create stage and prod evironments
Move config files (main.tf,outputs.tf,variables.tf) into 2
directory (prod,stage), added into these directory file terraform.tfvars.example
IP address in VPC module not changed because will be push to github
my own IP

loktionovam reviewed (commented):
Looks good to me

Durf1985 committed:
DO-61 Store a state file into S3 bucket
Created storage-bucket.tf in terraform directory
added backend.tf in directory:
* terraform/stage
* terraform/prod
* terraform

loktionovam commented on terraform/prod/variables.tf:
```diff
@@ -58,3 +58,9 @@ variable "db_disk_image" {
 }
+
+variable "source_ranges" {
+  type        = string
```
I think `list` type would be more suitable here

Durf1985 committed:
DO-62 Add provisioners to modules
Added `backend.tf` files to `prod` and `stage`
Added `storage-bucket.tf` configuration file
To remove the error of connecting the application to MongoDB,
you need to edit the mongod.conf configuration file on your DB VM instance
